Reply/app/src/main/java/com/example/reply/ui/ReplyListContent.kt

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2022 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.reply.ui
+
+import androidx.activity.compose.BackHandler
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.LazyListState
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Edit
+import androidx.compose.material3.Icon
+import androidx.compose.material3.LargeFloatingActionButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import com.example.reply.R
+import com.example.reply.data.Email
+import com.example.reply.ui.components.EmailDetailAppBar
+import com.example.reply.ui.components.ReplyEmailListItem
+import com.example.reply.ui.components.ReplyEmailThreadItem
+import com.example.reply.ui.components.ReplySearchBar
+import com.example.reply.ui.utils.ReplyContentType
+import com.example.reply.ui.utils.ReplyNavigationType
+
+@Composable
+fun ReplyInboxScreen(
+    contentType: ReplyContentType,
+    replyHomeUIState: ReplyHomeUIState,
+    navigationType: ReplyNavigationType,
+    closeDetailScreen: () -> Unit,
+    navigateToDetail: (Long) -> Unit
+) {
+    /**
+     * When moving from LIST page to LIST_AND_DETAIL page on screen expand, set first email selected as default
+     * When moving from LIST_AND_DETAIL page to LIST page clear the selection and user should see LIST screen.
+     */
+    LaunchedEffect(key1 = contentType) {
```

Comment:
I'm confused by this logic a little bit.

Right now, there's this behavior:
- Reading a detail while on a small screen
- Cause an activity recreation (resize, other config change)
- I am kicked out of the detail I was reading

In that case, I'd expect to continue reading the detail I was on.

Reply:
Ok I updated it to following -> 

- It will preserve the selected detail page from SINGLE to DUAL PANE
- From DUAL to SINGLE pane it will be reset to SINGLE pane(No detail) but maintaining the scroll state.

Reply:
I think the same issue is still present for activity recreation, since the `LaunchedEffect` isn't only running when the content type changes, it also runs on the first time (after recreation).

So if I'm reading a detail, then an activity recreation happens, the `LaunchedEffect` will run again and close the detail screen I was reading even if it remains in `SINGLE_PANE` the entire time. I think that should be fixable with some sort of additional `var previousContentType`?

Reply:
I used the replyHomeUIState's showSingleScreen to decide and it also survives configuration changes 👍🏽